clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -865,14 +897,15 @@ void EmitAssemblyHelper::EmitAssembly(BackendAction Action,
 
   std::unique_ptr<llvm::ToolOutputFile> ThinLinkOS, DwoOS;
 
+  // Clean-up SYCL device code if LLVM passes are disabled
+  if (LangOpts.SYCLIsDevice && CodeGenOpts.DisableLLVMPasses)
```

Comment:
More a suggestion for upstreaming purposes: `DisableLLVMPasses` is supposed to be a mode to retrieve LLVM IR as generated by clang before any LLVM optimizations (helpful for debugging). Perhaps it would be better to have a specific mode which would not run optimizations yet would run some clean-up passes.

Reply:
This change is just a refactoring of existing functionality.
@Naghasan, are you okay with addressing it in a separate PR?

Reply:
yes, it was more informative. As you refactored it, this was highlighted, but that's clearly out of scope.